magento/inventory
# 2170 reservation cli compensation

- Refactore given logic for collection inconsistencies
- Change CLI, which displays all given inconsistencies
- Introduce CLI command, which creates the compensations

### Description (*)
This PR changes the current implementation for 
- magento-engcom/msi#2140
- magento-engcom/msi#2169

and has dependency to PR https://github.com/magento-engcom/msi/issues/2171

### Fixed Issues (if relevant)
1. magento-engcom/msi#2170: Reservations CLI: Compensation

### Manual testing scenarios (*)
1. Create multiple orders 
2. Ship some orders and create an invoice (completed orders)
3. Remove reservations for DB 
4. Run `bin/magento inventory:reservation:list-inconsistencies` to see the given inconsistencies
5. Run `bin/magento inventory:reservation:create-compensations` to fix the reservation issues

### Contribution checklist (*)
